File: benchmarks/benchmark.py
import time
from clients.python.nktg_law.nktg import momentum, nktg1, nktg2

def run_benchmark():
    m, v, x, dm_dt = 5.0, 3.0, 2.0, 0.1
    iterations = 1000000

    start = time.time()
    for _ in range(iterations):
        p = momentum(m, v)
        a = nktg1(x, m, v)
        b = nktg2(dm_dt, m, v)
    end = time.time()

    print(f"Python benchmark: {iterations} iterations in {end - start:.2f} seconds")

if __name__ == "__main__":
    run_benchmark()

File: benchmarks/benchmark.java
import nktg.NKTgLaw;

public class benchmark {
    public static void main(String[] args) {
        double m = 5.0, v = 3.0, x = 2.0, dm_dt = 0.1;
        int iterations = 1_000_000;

        long start = System.nanoTime();
        for (int i = 0; i < iterations; i++) {
            double p = NKTgLaw.momentum(m, v);
            double NKTg1 = NKTgLaw.nktg1(x, m, v);
            double NKTg2 = NKTgLaw.nktg2(dm_dt, m, v);
        }
        long end = System.nanoTime();

        double elapsedSeconds = (end - start) / 1e9;
        System.out.printf("Java benchmark: %d iterations in %.2f seconds%n", iterations, elapsedSeconds);
    }
}
